doc-content/enterprise-only/installation/securing-passwords-jws-proc.adoc

```diff
@@ -0,0 +1,133 @@
+[id='securing-passwords-jws-proc_{context}']
+
+= Securing passwords with a keystore
+
+You can use a keystore to encrypt passwords that are used for communication between {CENTRAL} and {KIE_SERVER}. You should encrypt both controller and {KIE_SERVER} passwords. If {CENTRAL} and {KIE_SERVER} are deployed to different application servers then both application servers should use the keystore.
+
+Use Java Cryptography Extension KeyStore (JCEKS) for your keystore because it supports symmetric keys. Use KeyTool, which is part of the JDK installation, to create a new JCEKS.
+
+[NOTE]
+====
+If {KIE_SERVER} is not configured with JCEKS, {KIE_SERVER} passwords are stored in system properties in plain text form.
+====
+
+.Prerequisites
+
+* {KIE_SERVER} is installed in
+ifeval::["{context}" == "install-on-jws"]
+{JWS}.
+endif::[]
+ifeval::["{context}" == "install-on-tomcat"]
+{TOMCAT}.
+endif::[]
+* Java 8 or higher is installed.
+
+.Procedure
+ifeval::["{context}" == "install-on-jws"]
+. Open the `_TOMCAT_HOME_/tomcat/conf/tomcat-users.xml` file in a text editor.
+. Add a {KIE_SERVER} user with the `kie-server` role to the `_TOMCAT_HOME_/tomcat/conf/tomcat-users.xml` file. In the following example, replace `<USER>` and `<PASSWORD>` with the user and password of your choice.
++
+[source]
+----
+<role rolename="kie-server"/>
+<user username="<USER_NAME>" password="<PASSWORD>" roles="kie-server"/>
+----
+endif::[]
+
+ifeval::["{context}" == "install-on-tomcat"]
+. Open the `_TOMCAT_HOME_/tomcat/conf/tomcat-users.xml` file in a text editor.
+. Add a {KIE_SERVER} user with the `kie-server` role to the `_TOMCAT_HOME_/tomcat/conf/tomcat-users.xml` file. In the following example, replace `<USER>` and `<PASSWORD>` with the user and password of your choice.
++
+[source]
+----
+<role rolename="<kie-server>"/>
+<user username="<USER_NAME>" password="<PASSWORD>" roles="kie-server"/>
+----
+endif::[]
+
+
+. To use KeyTool to create a JCEKS, enter the following command in the Java 8 home directory:
++
+[source,bash]
+----
+$<JAVA_HOME>/bin/keytool -importpassword -keystore <KEYSTORE_PATH> -keypass <ALIAS_KEY_PASSWORD> -alias <PASSWORD_ALIAS> -storepass <KEYSTORE_PASSWORD> -storetype JCEKS
+----
++
+In this example, replace the following variables:
+
+* `<KEYSTORE_PATH>`: The path where the keystore will be stored
+* `<KEYSTORE_PASSWORD>`: The keystore password
+* `<ALIAS_KEY_PASSWORD>`: The password used to access values stored with the alias
+* `<PASSWORD_ALIAS>`: The alias of the entry to the process
+
+. When prompted, enter the password for the {KIE_SERVER} user that you created.
+
+. To set the system properties, complete one of these steps in the
+ifeval::["{context}" == "install-on-jws"]
+`_JWS_HOME_/tomcat/bin` directory
+endif::[]
+ifeval::["{context}" == "install-on-tomcat"]
+`_TOMCAT_HOME_/tomcat/bin` directory
+endif::[]
+and replace the variables as described in the table that follows:
```

Comment:
Can't look it up right now, but I think IBM would prefer "the following table:". I think I remember "follows:" being in the same discouraged camp as "the following:".  Two instances of this, so heads up.